x/concentrated-liquidity/swaps.go

```diff
@@ -365,8 +365,7 @@ func (k Keeper) computeOutAmtGivenIn(
 		} else if !sqrtPriceStart.Equal(sqrtPrice) {
 			// Otherwise if the sqrtPrice calculated from computeSwapStep does not equal the sqrtPrice we started with at the
 			// beginning of this iteration, we set the swapState tick to the corresponding tick of the sqrtPrice calculated from computeSwapStep
-			price := sqrtPrice.Mul(sqrtPrice)
-			swapState.tick, err = math.PriceToTickRoundDown(price, p.GetTickSpacing())
+			swapState.tick, err = math.SqrtPriceToTickRoundDownSpacing(sqrtPrice, p.GetTickSpacing())
```

Comment:
Unsure whether in/out of the scope of this PR but we should not be doing tick spacing rounding in swaps, only in LP logic to prevent initialization of ticks we do not desire.

It should be possible to be within tick spacing for the "current tick" upon swap completion. However, it should not be possible to initialize the tick at an index that is not a multiple of tick spacing.

Reply:
Hmm thanks for flagging this – not sure I fully follow the concern but sounds important. Let's discuss separately

Reply:
Not sure if I am following here, do you mean a pools current tick inter swap can be between tick spacings or the result can be between tick spacings? I would think at the very least the resulting tick once the swap is complete should be rounded.

Reply:
Yes, that's what I mean. Current tick can be between two initialized ticks inter-swap. That is, inside an active bucket. With tick rounding, the translation between current sqrt price and current tick would be broken.

I addressed this concern in this PR: https://github.com/osmosis-labs/osmosis/pull/5526 and left more detailed explanation.

Let's continue discussing this in #5526